src/sys/ptrace.rs

```diff
@@ -91,6 +92,15 @@ fn ptrace_peek(request: ptrace::PtraceRequest, pid: pid_t, addr: *mut c_void, da
     }
 }
 
+/// Function for ptrace requests that return values from the data field.
+fn ptrace_get_data<T>(request: ptrace::PtraceRequest, pid: pid_t) -> Result<T> {
+    let data: Box<T> = Box::new(unsafe { mem::uninitialized() });
```

Comment:
This seems pretty complicated. Please improve the doccomment and also add inline comments describing *why* it's doing things. It's rather hard to parse right now.